### PR TITLE
remove IDE specific files from .gitignore

### PR DESCRIPTION
**Pull request by Madhu-1**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -43,3 +43,8 @@
 *.un~
 Session.vim
 .netrwhist
+
+# Ignore vi backup files
```
Comment (msau42) [inline .gitignore]: @pohly brought this up in the kubernetes-csi standup meeting.  We think actually the .gitignore should only include files/binaries that are generated by this repository and not for files dependent on a dev's environment.

So for this particular repo, I think we should only ignore the bin/ directory

Comment (davidz627) [reply .gitignore]: https://github.com/kubernetes-csi/external-provisioner/pull/204#discussion_r245847823

**Comment by msau42** [comment]
@pohly I believe with your release-tools change we don't need this anymore?

**Comment by pohly** [comment]
"csi-release-tools" does not include .gitignore, therefore we still need this PR. However, we agreed to reduce .gitignore to just the repo-specific suppressions, so the PR needs to be updated.

@Madhu-1 can you do that?

**Comment by Madhu-1** [comment]
@pohly sure i will update this PR

**Comment by Madhu-1** [comment]
@pohly updated `.gitignore` PTAL

**Pull request merged**